src/components/accessories/admission/admissionForm/AdmissionForm.tsx

```diff
@@ -0,0 +1,238 @@
+import { useFormik } from "formik";
+import get from "lodash.get";
+import has from "lodash.has";
+import React, { FC, useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { object, string } from "yup";
+import warningIcon from "../../../../assets/warning-icon.png";
+import { AdmissionTypeDTO, DiseaseDTO, WardDTO } from "../../../../generated";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../../libraries/formDataHandling/functions";
+import { IState } from "../../../../types";
+import AutocompleteField from "../../autocompleteField/AutocompleteField";
+import ConfirmationDialog from "../../confirmationDialog/ConfirmationDialog";
+import DateField from "../../dateField/DateField";
+import SmallButton from "../../smallButton/SmallButton";
+import TextButton from "../../textButton/TextButton";
+import TextField from "../../textField/TextField";
+import "./styles.scss";
+import { AdmissionProps } from "./types";
+
+const AdmissionForm: FC<AdmissionProps> = ({
+  fields,
+  onSubmit,
+  submitButtonLabel,
+  resetButtonLabel,
+  isLoading,
+  shouldResetForm,
+  resetFormCallback,
+}) => {
+  const { t } = useTranslation();
+
+  const diagnosisInList = useSelector(
+    (state: IState) => state.diseases.diseasesIpdIn.data
+  );
+
+  const admissionTypes = useSelector(
+    (state: IState) => state.admissionTypes.allAdmissionTypes.data
+  );
+  const wards = useSelector((state: IState) => state.wards.allWards.data);
+
+  const toOptions = (
```

Comment:
Please rename this metho with a more specific name, as for eg: renderOptions or something like that.